pkg/ebpf/c/bpf_telemetry.h

```diff
@@ -14,33 +20,31 @@ BPF_HASH_MAP(helper_err_telemetry_map, unsigned long, helper_err_telemetry_t, 25
-                }                                                                  \
-                errno_slot &= (T_MAX_ERRNO - 1);                                   \
-                long *target = &entry->err_count[errno_slot];                      \
-                unsigned long add = 1;                                             \
-                /* Patched instruction for 4.14+: __sync_fetch_and_add(target, 1);
-                 * This patch point is placed here because the above instruction
-                 * fails on the 4.4 verifier. On 4.4 this instruction is replaced
-                 * with a nop: r1 = r1 */                                          \
-                bpf_telemetry_update_patch((unsigned long)target, add);            \
-            }                                                                      \
-        }                                                                          \
-        errno_ret;                                                                 \
-    })
+#define __record_map_telemetry(map, errno_ret) \
+    long errno_slot;                                                           \
+    unsigned long err_telemetry_key;                                           \
+    LOAD_CONSTANT(MK_KEY(map), err_telemetry_key);                             \
+    if (err_telemetry_key > 0) {                              \
+        map_err_telemetry_t *entry =                                           \
+        bpf_map_lookup_elem(&map_err_telemetry_map, &err_telemetry_key);       \
+        if (entry) {                                                           \
+            errno_slot = errno_ret * -1;                                       \
+            if (errno_slot >= T_MAX_ERRNO) {                                   \
+                errno_slot = T_MAX_ERRNO - 1;                                  \
+                errno_slot &= (T_MAX_ERRNO - 1);                               \
+            }                                                                  \
+            errno_slot &= (T_MAX_ERRNO - 1);                                   \
+            long *target = &entry->err_count[errno_slot];                      \
+            unsigned long add = 1;                                             \
+            /* Patched instruction for 4.14+: __sync_fetch_and_add(target, 1);
+             * This patch point is placed here because the above instruction
+             * fails on the 4.4 verifier. On 4.4 this instruction is replaced
+             * with a nop: r1 = r1 */                                          \
+            bpf_telemetry_update_patch((unsigned long)target, add);            \
+        }                                                                      \
+    }                                                                          \
+
+#define map_update_with_telemetry(fn, map, args...) \
```

Comment:
Why are we keeping this macro around? It is empty?

Reply:
Yeah. I meant to remove.